### PR TITLE
Initial implementation of the Operator Interface class

### PR DESCRIPTION
**Pull request by zbrotherton**

### CODE REVIEW

File: 2022Swerve/src/main/java/frc/robot/OI.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            this.yIntercept = yIntercept;
+            this.curvature = curvature;
+            this.deadzone = deadzone;
+        }
+
+        public double calculate(double input){
+            return Math.abs(input) <  deadzone ? 0 : 
+            ySaturation * (Math.signum(input) * yIntercept + 
+            (1 - yIntercept) * (curvature * Math.pow(input, 3) +
+            (1 - curvature) * input));
```
Comment (azvanderpas) [inline 2022Swerve/src/main/java/frc/robot/OI.java:82]: Comments showing a prettier version of the equation your using, and short explanation of it should get added.

File: 2022Swerve/src/main/java/frc/robot/OI.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        public static JoystickButton getZeroButton(){
+            return new JoystickButton(kJoystick, kZeroButtonID);
+        }
+    }
+
+    public static final class Operator{
+
+    }
```
Comment (azvanderpas) [inline 2022Swerve/src/main/java/frc/robot/OI.java:49]: Is this class needed?

File: 2022Swerve/src/main/java/frc/robot/subsystems/SwerveSubsystem.java [hunk 3]
```diff
@@ -132,12 +113,40 @@ public SwerveModuleState[] convertToModuleStates(double xTranslation, double yTr
...
+        double rotationalK = Math.abs(speeds.omegaRadiansPerSecond) / DriveConstants.kPhysicalMaxAngularSpeed;
+        double k = Math.max(translationalK, rotationalK);
+      
+        // Find the how fast the fastest spinning drive motor is spinning                                       
+        double realMaxSpeed = 0.0;
+        for (SwerveModuleState moduleState : desiredStates) {
+          realMaxSpeed = Math.max(realMaxSpeed, Math.abs(moduleState.speedMetersPerSecond));
+        }
+      
+        double scale = Math.min(k * DriveConstants.kPhysicalMaxSpeed / realMaxSpeed, 1);
```
Comment (azvanderpas) [inline 2022Swerve/src/main/java/frc/robot/subsystems/SwerveSubsystem.java]: Does ```kPhysicalMaxSpeed``` make sense here even in the case that ```k``` comes from ```rotationalK```?
This whole method could use some comments about what math is being done here

File: 2022Swerve/src/main/java/frc/robot/subsystems/SwerveSubsystem.java [hunk 4]
```diff
@@ -132,12 +113,40 @@ public SwerveModuleState[] convertToModuleStates(double xTranslation, double yTr
...
+        for (SwerveModuleState moduleState : desiredStates) {
+          moduleState.speedMetersPerSecond *= scale;
+        }
+    }
+
     public void setModuleStates(SwerveModuleState[] desiredStates) {
-        SwerveDriveKinematics.desaturateWheelSpeeds(desiredStates, DriveConstants.kPhysicalMaxSpeed);
+        this.normalizeDrive(desiredStates, DriveConstants.kDriveKinematics.toChassisSpeeds(desiredStates));
```
Comment (azvanderpas) [inline 2022Swerve/src/main/java/frc/robot/subsystems/SwerveSubsystem.java]: A comment for future reference about why ```desatureWheelSpeeds``` isn't sufficient here would probably be good.

**Review by azvanderpas — CHANGES_REQUESTED** [review]
Mainly just want some more documentation on some of the math. Then this is good to go

File: 2022Swerve/src/main/java/frc/robot/OI.java (hunk 1 above)
Comment (zbrotherton) [reply 2022Swerve/src/main/java/frc/robot/OI.java:82]: Further explanation on control curves, how they work, and an online graphing calculator representation are all in the swerve research document, but I did add an explanation of what each variable does, and wrote a more understandable version of the equation within the comments

File: 2022Swerve/src/main/java/frc/robot/OI.java (hunk 2 above)
Comment (zbrotherton) [reply 2022Swerve/src/main/java/frc/robot/OI.java:49]: The class is there to show intention of containing controls for the operator within a separate class

**Review by adamzvp — APPROVED** [review]

**Review by azvanderpas — APPROVED** [review]

**Pull request merged**